### PR TITLE
Improve ServerGUI

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**
Changes:
- Added TPS entry under avera ms
- Added paper logo as frame icon
- Small code tweaks

I think i want to fix a small issue here too, but im not sure where its caused:
paper

May someone help me there?

### CODE REVIEW

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+-import javax.swing.JList;
+-import javax.swing.JPanel;
+-import javax.swing.JScrollBar;
+-import javax.swing.JScrollPane;
+-import javax.swing.JTextArea;
+-import javax.swing.JTextField;
+-import javax.swing.SwingUtilities;
+-import javax.swing.UIManager;
++import javax.imageio.ImageIO;
++import javax.swing.*;
```
Comment (Proximyst) [inline Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: Don't do this to the poor imports.

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+     private static final Logger LOGGER = LogManager.getLogger();
+     private final DedicatedServer c;
+     private Thread d;
+@@ -42,26 +36,34 @@ public class ServerGUI extends JComponent {
+     public static ServerGUI a(final DedicatedServer dedicatedserver) {
+         try {
+             UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+-        } catch (Exception exception) {
+-            ;
++        } catch (Exception ignored) { // Paper
```
Comment (Proximyst) [inline Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: Nothing changed?

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+         } else {
+             Document document = jtextarea.getDocument();
+             JScrollBar jscrollbar = jscrollpane.getVerticalScrollBar();
+@@ -175,8 +175,7 @@ public class ServerGUI extends JComponent {
+ 
+             try {
+                 document.insertString(document.getLength(), ANSI.matcher(s).replaceAll(""), (AttributeSet) null); // CraftBukkit
+-            } catch (BadLocationException badlocationexception) {
+-                ;
++            } catch (BadLocationException ignore) { // Paper
```
Comment (Proximyst) [inline Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: Nothing changed.

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+         final JFrame jframe = new JFrame("Minecraft server");
+         final ServerGUI servergui = new ServerGUI(dedicatedserver);
+ 
+-        jframe.setDefaultCloseOperation(2);
++        jframe.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE); // Paper
+         jframe.add(servergui);
+         jframe.pack();
+         jframe.setLocationRelativeTo((Component) null);
+         jframe.setVisible(true);
++        jframe.setName("Minecraft server"); // Paper
```
Comment (Proximyst) [inline Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: This is likely the cause to your issue.

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
++            jframe.setIconImage(ImageIO.read(Objects.requireNonNull(ServerGUI.class.getClassLoader().getResourceAsStream("logo.png"))));
++        } catch (IOException ignore) {
++        }
++        // Paper end
++
+         jframe.addWindowListener(new WindowAdapter() {
+             public void windowClosing(WindowEvent windowevent) {
+                 if (!servergui.f.getAndSet(true)) {
+                     jframe.setTitle("Minecraft server - shutting down!");
++                    jframe.setName("Minecraft server - shutting down!"); // Paper
```
Comment (Proximyst) [inline Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: Setting name is unnecessary, and above.

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+                     dedicatedserver.safeShutdown(true);
+                     servergui.f();
+                 }
+-
```
Comment (Proximyst) [inline Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: Unnecessary diff.

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+         jtextarea.setEditable(false);
+         jtextarea.setFont(ServerGUI.a);
+@@ -161,9 +163,7 @@ public class ServerGUI extends JComponent {
+     private static final java.util.regex.Pattern ANSI = java.util.regex.Pattern.compile("\\x1B\\[([0-9]{1,2}(;[0-9]{1,2})*)?[m|K]"); // CraftBukkit
+     public void a(JTextArea jtextarea, JScrollPane jscrollpane, String s) {
+         if (!SwingUtilities.isEventDispatchThread()) {
+-            SwingUtilities.invokeLater(() -> {
+-                this.a(jtextarea, jscrollpane, s);
+-            });
++            SwingUtilities.invokeLater(() -> this.a(jtextarea, jscrollpane, s)); // Paper
```
Comment (Proximyst) [inline Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: Literally nothing changed here from what I can see?

File: Spigot-Server-Patches/0581-Add-tps-to-gui.patch [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+     public void update() {
+         GraphData data = RAMGraph.DATA.peekLast();
+         Vector<String> vector = new Vector<>();
++
++        // Paper start - Add tps entry
++        double tps = Bukkit.getTPS()[0];
+         vector.add("Memory use: " + (data.getUsedMem() / 1024L / 1024L) + " mb (" + (data.getFree() * 100L / data.getMax()) + "% free)");
+         vector.add("Heap: " + (data.getTotal() / 1024L / 1024L) + " / " + (data.getMax() / 1024L / 1024L) + " mb");
+         vector.add("Avg tick: " + DECIMAL_FORMAT.format(getAverage(server.getTickTimes())) + " ms");
++        vector.add("TPS: " + ( ( tps > 21.0 ) ? "*" : "" ) + Math.min( Math.round( tps * 100.0 ) / 100.0, 20.0 ));
```
Comment (Proximyst) [inline Spigot-Server-Patches/0581-Add-tps-to-gui.patch]: Perhaps rather print it like with /tps?

**Review by Proximyst — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch (hunk 4 above)
Comment (AlexProgrammerDE) [reply Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: You mean setting the name?

File: Spigot-Server-Patches/0581-Add-tps-to-gui.patch (hunk 8 above)
Comment (AlexProgrammerDE) [reply Spigot-Server-Patches/0581-Add-tps-to-gui.patch]: ok, but i thought it might look a little bit too large for smaller scaled windows

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch (hunk 2 above)
Comment (AlexProgrammerDE) [reply Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: I looked kinda messy to me, but if thats bad imma revert it.

Comment (AlexProgrammerDE) [reply Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: the constants are useful in case the library decides to change some values and makes the code way more readable since the are a small description of what they are. Just numbers would require someone to manually search for the value of it.

File: Spigot-Server-Patches/0582-Add-ServerGUI-image.patch (hunk 1 above)
Comment (AlexProgrammerDE) [reply Spigot-Server-Patches/0582-Add-ServerGUI-image.patch]: Thats happened somewhere in my IDE. Imma revert that...

File: Spigot-Server-Patches/0581-Add-tps-to-gui.patch (hunk 8 above)
Comment (AlexProgrammerDE) [reply Spigot-Server-Patches/0581-Add-tps-to-gui.patch]: also that code is actually from the tps command. I just didn't really want to make it look too large
